### PR TITLE
Implements localization for ModifierType

### PR DESCRIPTION
**Pull request by imfanhua**
Implements localization on ModifierType, but some functions or the placement of translation entries may need to be adjusted.

Changes:
- Implements localization on ModifierType
- Added modifier types EN locale

### CODE REVIEW

**Comment by Somod1** [comment]
烦华牛逼 加接口了

**Comment by CodeTappert** [comment]
Please add also the files for all other languages. (just copy the english one). 

I think otherwise there will be problems when someone starts the game with a non-english language

**Comment by imfanhua** [comment]
> Please add also the files for all other languages. (just copy the english one).
> 
> I think otherwise there will be problems when someone starts the game with a non-english language

Okay I can do that, btw I tested it in Simplified Chinese and there were no issues

**Comment by imfanhua** [comment]
Done, awaiting review and approval.

**Comment by CodeTappert** [comment]
So i already want to prepare the german version of this. 

But i am wondering. Why do some of the descriptions end with a "." but others dont?

File: src/locales/de/modifier-type.ts [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+  },
+  EvolutionItem: {
+    "NONE": "None",
+
+    "LINKING_CORD": "Linking Cord",
+    "SUN_STONE": "Sun Stone",
+    "MOON_STONE": "Moon Stone",
+    "LEAF_STONE": "Leaf Stone",
+    "FIRE_STONE": "Fire Stone",
+    "WATER_STONE": "WaterS tone",
```
Comment (CodeTappert) [inline src/locales/de/modifier-type.ts:280]: Error here. (in all languages)

Comment (imfanhua) [reply src/locales/de/modifier-type.ts]: Thx, fixed now

**Comment by CodeTappert** [comment]
Here is the german localization:

```ts
import { ModifierTypeTranslationEntries } from "#app/plugins/i18n";

export const modifierType: ModifierTypeTranslationEntries = {
  ModifierType: {
    "AddPokeballModifierType": {
      name: "{{modifierCount}}x {{pokeballName}}",
      description: "Erhalte {{pokeballName}} x{{modifierCount}} (Inventar: {{pokeballAmount}}) \nFangrate: {{catchRate}}",
    },
    "AddVoucherModifierType": {
      name: "{{modifierCount}}x {{voucherTypeName}}",
      description: "Erhalte {{voucherTypeName}} x{{modifierCount}}",
    },
    "PokemonHeldItemModifierType": {
      extra: {
        "inoperable": "{{pokemonName}} kann dieses\nItem nicht nehmen!",
        "tooMany": "{{pokemonName}} hat zu viele\nvon diesem Item!",
      }
    },
    "PokemonHpRestoreModifierType": {
      description: "Füllt {{restorePoints}} KP oder {{restorePercent}}% der KP für ein Pokémon auf. Je nachdem, welcher Wert höher ist",
      extra: {
        "fully": "Füllt die KP eines Pokémon wieder vollständig auf.",
        "fullyWithStatus": "Füllt die KP eines Pokémon wieder vollständig auf und behebt alle Statusprobleme",
      }
    },
    "PokemonReviveModifierType": {
      description: "Belebt ein kampunfähiges Pokémon wieder und stellt {{restorePercent}}% KP wieder her",
    },
    "PokemonStatusHealModifierType": {
      description: "Behebt alle Statusprobleme eines Pokémon",
    },
    "PokemonPpRestoreModifierType": {
      description: "Füllt {{restorePoints}} AP der ausgewählten Attacke eines Pokémon auf",
      extra: {
        "fully": "Füllt alle AP der ausgewählten Attacke eines Pokémon auf",
      }
    },
    "PokemonAllMovePpRestoreModifierType": {
      description: "Stellt {{restorePoints}} AP für alle Attacken eines Pokémon auf",
      extra: {
        "fully": "Füllt alle AP für alle Attacken eines Pokémon auf",
      }
    },
    "PokemonPpUpModifierType": {
      description: "Erhöht die maximale Anzahl der AP der ausgewählten Attacke eines Pokémon um {{upPoints}} für jede 5 maximale AP (maximal 3)",
    },
    "PokemonNatureChangeModifierType": {
      name: "{{natureName}} Minze",
      description: "Ändert das Wesen eines Pokémon zu {{natureName}} und schaltet dieses Wesen permanent für diesen Starter frei.",
    },
    "DoubleBattleChanceBoosterModifierType": {
      description: "Verdoppelt die Wahrscheinlichkeit, dass die nächsten {{battleCount}} Begegnungen mit wilden Pokémon ein Doppelkampf sind.",
    },
    "TempBattleStatBoosterModifierType": {
      description: "Erhöht die {{tempBattleStatName}} aller Teammitglieder für 5 Kämpfe um eine Stufe",
    },
    "AttackTypeBoosterModifierType": {
      description: "Erhöht die Stärke aller {{moveType}}attacken eines Pokémon um 20%",
    },
    "PokemonLevelIncrementModifierType": {
      description: "Erhöht das Level eines Pokémon um 1",
    },
    "AllPokemonLevelIncrementModifierType": {
      description: "Erhöht das Level aller Teammitglieder um 1",
    },
    "PokemonBaseStatBoosterModifierType": {
      description: "Erhöht den {{statName}} Basiswert des Trägers um 10%. Das Stapellimit erhöht sich, je höher dein IS-Wert ist.",
    },
    "AllPokemonFullHpRestoreModifierType": {
      description: "Stellt 100% der KP aller Pokémon her",
    },
    "AllPokemonFullReviveModifierType": {
      description: "Belebt alle kampunfähigen Pokémon wieder und stellt ihre KP vollständig wieder her",
    },
    "MoneyRewardModifierType": {
      description:"Gewährt einen {{moneyMultiplier}} Geldbetrag von (₽{{moneyAmount}})",
      extra: {
        "small": "kleinen",
        "moderate": "moderaten",
        "large": "großen",
      },
    },
    "ExpBoosterModifierType": {
      description: "Erhöht die erhaltenen Erfahrungspunkte um {{boostPercent}}%",
    },
    "PokemonExpBoosterModifierType": {
      description: "Erhöht die Menge der erhaltenen Erfahrungspunkte für den Träger um {{boostPercent}}%",
    },
    "PokemonFriendshipBoosterModifierType": {
      description: "Erhöht den Freundschaftszuwachs pro Sieg um 50%.",
    },
    "PokemonMoveAccuracyBoosterModifierType": {
      description: "Erhöht die Genauigkeit der Angriffe um {{accuracyAmount}} (maximal 100)",
    },
    "PokemonMultiHitModifierType": {
      description: "Attacken treffen ein weiteres mal mit einer Reduktion von 60/75/82,5% der Stärke",
    },
    "TmModifierType": {
      name: "TM{{moveId}} - {{moveName}}",
      description: "Bringt einem Pokémon {{moveName}} bei",
    },
    "EvolutionItemModifierType": {
      description: "Erlaubt es bestimmten Pokémon sich zu entwickeln",
    },
    "FormChangeItemModifierType": {
      description: "Erlaubt es bestimmten Pokémon ihre Form zu ändern",
    },
    "FusePokemonModifierType": {
      description: "Fusioniert zwei Pokémon (überträgt die Fähigkeit, teilt Basiswerte und Typ auf, gemeinsamer Attackenpool),
    },
    "TerastallizeModifierType": {
      name: "{{teraType}} Terra-Stück",
      description: "{{teraType}} Terakristallisiert den Träger für bis zu 10 Kämpfe",
    },
    "ContactHeldItemTransferChanceModifierType": {
      description:"Beim Angriff besteht eine {{chancePercent}}%ige Chance, dass das getragene Item des Gegners gestohlen wird."
    },
    "TurnHeldItemTransferModifierType": {
      description: "Jede Runde erhält der Träger ein getragenes Item des Gegners",
    },
    "EnemyAttackStatusEffectChanceModifierType": {
      description:  "Fügt Angriffen eine {{chancePercent}}%ige Chance hinzu, {{statusEffect}} zu verursachen",
    },
    "EnemyEndureChanceModifierType": {
      description: "Gibt den Träger eine {{chancePercent}}%ige Chance, einen Angriff zu überleben",
    },

    "RARE_CANDY": { name: "Sonderbonbon" },
    "RARER_CANDY": { name: "Supersondererbonbon" },

    "MEGA_BRACELET": { name: "Mega-Armband", description: "Mega-Steine werden verfügbar" },
    "DYNAMAX_BAND": { name: "Dynamax-Band", description: "Dyna-Pilze werden verfügbar" },
    "TERA_ORB": { name: "Terakristall-Orb", description: "Tera-Stücke werden verfügbar" },

    "MAP": { name: "Karte", description: "Ermöglicht es dir, an einer Kreuzung dein Ziel zu wählen." },

    "POTION": { name: "Trank" },
    "SUPER_POTION": { name: "Supertrank" },
    "HYPER_POTION": { name: "Hypertrank" },
    "MAX_POTION": { name: "Top-Trank" },
    "FULL_RESTORE": { name: "Top-Genesung" },

    "REVIVE": { name: "Beleber" },
    "MAX_REVIVE": { name: "Top-Beleber" },

    "FULL_HEAL": { name: "Hyperheiler" },

    "SACRED_ASH": { name: "Zauberasche" },

    "REVIVER_SEED": { name: "Belebersamen", description: "Belebt den Träger mit der Hälfte seiner KP wieder sollte er kampfunfähig werden" },

    "ETHER": { name: "Äther" },
    "MAX_ETHER": { name: "Top-Äther" },

    "ELIXIR": { name: "Elixir" },
    "MAX_ELIXIR": { name: "Top-Elixir" },

    "PP_UP": { name: "AP-Plus" },
    "PP_MAX": { name: "AP-Top" },

    "LURE": { name: "Lockparfüm" },
    "SUPER_LURE": { name: "Super-Lockparfüm" },
    "MAX_LURE": { name: "Top-Lockparfüm" },

    "MEMORY_MUSHROOM": { name: "Erinnerungspilz", description: "Lässt ein Pokémon eine vergessene Attacke wiedererlernen" },

    "EXP_SHARE": { name: "EP-Teiler", description: "Pokémon, die nicht am Kampf teilgenommen haben, bekommen 20% der Erfahrungspunkte eines Kampfteilnehmers" },
    "EXP_BALANCE": { name: "EP-Ausgleicher", description: "Gewichtet die in Kämpfen erhaltenen Erfahrungspunkte auf niedrigstufigere Gruppenmitglieder." },

    "OVAL_CHARM": { name: "Ovalpin", description: "Wenn mehrere Pokémon am Kampf teilnehmen, erhählt jeder von ihnen 10% extra Erfahrungspunkte" },

    "EXP_CHARM": { name: "EP-Pin" },
    "SUPER_EXP_CHARM": { name: "Super-EP-Pin" },
    "GOLDEN_EXP_CHARM": { name: "Goldener EP-Pin" },

    "LUCKY_EGG": { name: "Glücks-Ei" },
    "GOLDEN_EGG": { name: "Goldenes Ei" },

    "SOOTHE_BELL": { name: "Sanftglocke" },

    "SOUL_DEW": { name: "Seelentau", description: "Erhöht den Einfluss des Wesens eines Pokemon auf seine Werte um 10% (additiv)" },

    "NUGGET": { name: "Nugget" },
    "BIG_NUGGET": { name: "Riesennugget" },
    "RELIC_GOLD": { name: "Alter Dukat" },

    "AMULET_COIN": { name: "Münzamulett", description: "Erhöht das Preisgeld um 20%" },
    "GOLDEN_PUNCH": { name: "Goldschlag", description: "Gewährt Geld in Höhe von 50% des zugefügten Schadens" },
    "COIN_CASE": { name: "Münzkorb", description: "Erhalte nach jedem 10ten Kampf 10% Zinsen auf dein Geld" },

    "LOCK_CAPSULE": { name: "Tresorkapsel", description: "Erlaubt es die Seltenheitsstufe der Items festzusetzen wenn diese neu gerollt werden" },

    "GRIP_CLAW": { name: "Griffklaue" },
    "WIDE_LENS": { name: "Großlinse" },

    "MULTI_LENS": { name: "Mehrfachlinse" },

    "HEALING_CHARM": { name: "Heilungspin", description: "Erhöht die Effektivität von KP-Wiederherstellungsattacken sowie Items um 10% (Beleber ausgenommen)" },
    "CANDY_JAR": { name: "Bonbonglas", description: "Erhöht die Anzahl der Level die ein Sonderbonbon erhöht um 1" },

    "BERRY_POUCH": { name: "Beerentüte", description: "Fügt eine 25% Chance hinzu, dass Beeren nicht verbraucht werden" },

    "FOCUS_BAND": { name: "Fokusband", description: "Fügt eine 10% Chance hinzu, dass Angriffe die zur Kampfunfähigkeit führen mit 1 KP überlebt werden" },

    "KINGS_ROCK": { name: "King-Stein", description: "Fügt eine 10% Chance hinzu, dass der Gegner nach einem Angriff zurückschreckt" },

    "LEFTOVERS": { name: "Überreste", description: "Heilt 1/16 der maximalen KP eines Pokémon pro Runde" },
    "SHELL_BELL": { name: "Muschelglocke", description: "Heilt den Anwender um 1/8 des von ihm zugefügten Schadens" },

    "BATON": { name: "Stab", description: "Ermöglicht das Weitergeben von Effekten beim Wechseln von Pokémon, wodurch auch Fallen umgangen werden." },

    "SHINY_CHARM": { name: "Schillerpin", description: "Erhöht die Chance deutlich, dass ein wildes Pokémon ein schillernd ist" },
    "ABILITY_CHARM": { name: "Ability Charm", description: "Erhöht die Chance deutlich, dass ein wildes Pokémon eine versteckte Fähigkeit hat" },

    "IV_SCANNER": { name: "IS-Scanner", description: "Erlaubt es die IS-Werte von wilden Pokémon zu scannen. 2 IS-Werte werden pro Staplung angezeigt. Die besten IS-Werte werden zuerst gezeigt" },

    "DNA_SPLICERS": { name: "DNS-Keil" },

    "MINI_BLACK_HOLE": { name: "Mini schwarzes Loch" },

    "GOLDEN_POKEBALL": { name: "Goldener Pokéball", description: "Fügt eine zusätzliche Item-Auswahlmöglichkeit nach jedem Kampf hinzu" },

    "ENEMY_DAMAGE_BOOSTER": { name: "Schadensmarke", description: "Erhöht den Schaden um 5%" },
    "ENEMY_DAMAGE_REDUCTION": { name: "Schutzmarke", description: "Verringert den erhaltenen Schaden um 2,5%" },
    "ENEMY_HEAL": { name: "Wiederherstellungsmarke", description: "Heilt 2% der maximalen KP pro Runde" },
    "ENEMY_ATTACK_POISON_CHANCE": { name: "Giftmarke" },
      "ENEMY_ATTACK_PARALYZE_CHANCE": { "name": "Lähmungsmarke" },
      "ENEMY_ATTACK_SLEEP_CHANCE": { "name": "Schlafmarke" },
      "ENEMY_ATTACK_FREEZE_CHANCE": { "name": "Gefriermarke" },
      "ENEMY_ATTACK_BURN_CHANCE": { "name": "Brandmarke" },
      "ENEMY_STATUS_EFFECT_HEAL_CHANCE": { "name": "Vollheilungsmarke", "description": "Fügt eine 10%ige Chance hinzu, jede Runde einen Statuszustand zu heilen" },
      "ENEMY_ENDURE_CHANCE": { "name": "Ausdauer-Marke" },
      "ENEMY_FUSED_CHANCE": { "name": "Fusionsmarke", "description": "Fügt eine 1%ige Chance hinzu, dass ein wildes Pokémon eine Fusion ist" },

  },
  TempBattleStatBoosterItem: {
    "x_attack": "X-Angriff",
    "x_defense": "X-Verteidigung",
    "x_sp_atk": "X-Sp.-Ang.",
      "x_sp_def": "X-Sp.-Vert.",
    "x_speed": "X-Tempo",
    "x_accuracy": "X-Treffer",
    "dire_hit": "X-Volltreffer",
  },
  AttackTypeBoosterItem: {
    "silk_scarf": "Seidenschal",
    "black_belt": "Schwarzgurt",
    "sharp_beak": "Spitzer Schnabel",
    "poison_barb": "Giftstich",
    "soft_sand": "Pudersand",
    "hard_stone": "Granitstein",
    "silver_powder": "Silberstaub",
    "spell_tag": "Bannsticker",
    "metal_coat": "Metallmantel",
    "charcoal": "Holzkohle",
    "mystic_water": "Zauberwasser",
    "miracle_seed": "Wundersaat",
    "magnet": "Magnet",
    "twisted_spoon": "Krümmlöffel",
    "never-nelt_ice": "Ewiges Eis",
    "dragon_fang": "Drachenzahn",
    "black_glasses": "Schattenbrille",
    "fairy_feather": "Feendaune",
  },
  BaseStatBoosterItem: {
    "hp_up": "KP-Plus",
    "protein": "Protein",
    "iron": "Eisen",
    "calcium": "Kalzium",
    "zinc": "Zink",
    "carbos": "Carbon",
  },
  EvolutionItem: {
    "NONE": "Keins",

    "LINKING_CORD": "Linkkabel",
    "SUN_STONE": "Sonnenstein",
    "MOON_STONE": "Mondstein",
    "LEAF_STONE": "Blattstein",
    "FIRE_STONE": "Feuerstein",
    "WATER_STONE": "Wasserstein",
    "THUNDER_STONE": "Donnerstein",
    "ICE_STONE": "Eisstein",
    "DUSK_STONE": "Finsterstein",
    "DAWN_STONE": "Funkelstein",
    "SHINY_STONE": "Leuchtstein",
    "CRACKED_POT": "Rissige Kanne",
    "SWEET_APPLE": "Süßer Apfel",
    "TART_APPLE": "Saurer Apfel",
    "STRAWBERRY_SWEET": "Zucker-Erdbeere",
    "UNREMARKABLE_TEACUP": "Simple Teeschale",

    "CHIPPED_POT": "Löchrige Kanne",
    "BLACK_AUGURITE": "Schwarzaugit",
    "GALARICA_CUFF": "Galarnuss-Reif",
    "GALARICA_WREATH": "Galarnuss-Kranz",
    "PEAT_BLOCK": "Torfblock",
    "AUSPICIOUS_ARMOR": "Glorienrüstung",
    "MALICIOUS_ARMOR": "Fluchrüstung",
    "MASTERPIECE_TEACUP": "Edle Teeschale",
    "METAL_ALLOY": "Legierungsmetall",
    "SCROLL_OF_DARKNESS": "Unlicht-Schriftrolle",
    "SCROLL_OF_WATERS": "Wasser-Schriftrolle",
    "SYRUPY_APPLE": "Saftiger Apfel",
  },
  FormChangeItem: {
    "NONE": "Keins",

    "ABOMASITE": "Rexblisarnit",
    "ABSOLITE": "Absolnit",
    "AERODACTYLITE": "Aerodactylonit",
    "AGGRONITE": "Stollossnit",
    "ALAKAZITE": "Simsalanit",
    "ALTARIANITE": "Altarianit",
    "AMPHAROSITE": "Ampharosnit",
    "AUDINITE": "Ohrdochnit",
    "BANETTITE": "Banetteonit",
    "BEEDRILLITE": "Bibornit",
    "BLASTOISINITE": "Turtoknit",
    "BLAZIKENITE": "Lohgocknit",
    "CAMERUPTITE": "Cameruptnit",
    "CHARIZARDITE_X": "Gluraknit X",
    "CHARIZARDITE_Y": "Gluraknit Y",
    "DIANCITE": "Diancienit",
    "GALLADITE": "Galagladinit",
    "GARCHOMPITE": "Knakracknit",
    "GARDEVOIRITE": "Guardevoirnit",
    "GENGARITE": "Gengarnit ",
    "GLALITITE": "Firnontornit",
    "GYARADOSITE": "Garadosnit",
    "HERACRONITE": "Skarabornit",
    "HOUNDOOMINITE": "Hundemonit ",
    "KANGASKHANITE": "Kangamanit",
    "LATIASITE": "Latiasnit",
    "LATIOSITE": "Latiosnit",
    "LOPUNNITE": "Schlapornit",
    "LUCARIONITE": "Lucarionit",
    "MANECTITE": "Voltensonit",
    "MAWILITE": "Flunkifernit",
    "MEDICHAMITE": "Meditalisnit",
    "METAGROSSITE": "Metagrossnit",
    "MEWTWONITE_X": "Mewtunit X",
    "MEWTWONITE_Y": "Mewtunit Y",
    "PIDGEOTITE": "Taubossnit",
    "PINSIRITE": "Pinsirnit",
    "RAYQUAZITE": "Rayquazanit",
    "SABLENITE": "Zobirisnit",
    "SALAMENCITE": "Brutalandanit",
    "SCEPTILITE": "Gewaldronit",
    "SCIZORITE": "Scheroxnit",
    "SHARPEDONITE": "Tohaidonit",
    "SLOWBRONITE": "Lahmusnit",
    "STEELIXITE": "Stahlosnit",
    "SWAMPERTITE": "Sumpexnit",
    "TYRANITARITE": "Despotarnit",
    "VENUSAURITE": "Bisaflornit",

    "BLUE_ORB": "Blauer Edelstein",
    "RED_ORB": "Roter Edelstein",
    "SHARP_METEORITE": "Scharfer Meteorit",
    "HARD_METEORITE": "Harter Meteorit",
    "SMOOTH_METEORITE": "Glatter Meteorit",
    "ADAMANT_CRYSTAL": "Adamantkristall",
    "LUSTROUS_ORB": "Weiß-Orb",
    "GRISEOUS_CORE": "Platinumkristall",
    "REVEAL_GLASS": "Wahrspiegel",
    "GRACIDEA": "Gracidea",
    "MAX_MUSHROOMS": "Dyna-Pilz",
    "DARK_STONE": "Dunkelstein",
    "LIGHT_STONE": "Lichtstein",
    "PRISON_BOTTLE": "Banngefäß",
    "N_LUNARIZER": "Necrolun",
    "N_SOLARIZER": "Necrosol",
    "RUSTED_SWORD": "Rostiges Schwert",
    "RUSTED_SHIELD": "Rostiges Schild",
    "ICY_REINS_OF_UNITY": "eisige Zügel des Bundes",
    "SHADOW_REINS_OF_UNITY": "schattige Zügel des Bundes",
    "WELLSPRING_MASK": "Brunnenmaske",
    "HEARTHFLAME_MASK": "Ofenmaske",
    "CORNERSTONE_MASK": "Fundamentmaske",
    "SHOCK_DRIVE": "Blitzmodul",
    "BURN_DRIVE": "Flammenmodul",
    "CHILL_DRIVE": "Gefriermodul",
    "DOUSE_DRIVE": "Aquamodul",
  },
  TeraType: {
    "UNKNOWN": "Unbekannt",
    "NORMAL": "Normal",
    "FIGHTING": "Kampf",
    "FLYING": "Flug",
    "POISON": "Gift",
    "GROUND": "Boden",
    "ROCK": "Gestein",
    "BUG": "Käfer",
    "GHOST": "Geist",
    "STEEL": "Stahl",
    "FIRE": "Feuer",
    "WATER": "Wasser",
    "GRASS": "Pflanze",
    "ELECTRIC": "Elektro",
    "PSYCHIC": "Psycho",
    "ICE": "Eis",
    "DRAGON": "Drache",
    "DARK": "Unlicht",
    "FAIRY": "Fee",
    "STELLAR": "Stellar",
  },
} as const;
```

**Comment by imfanhua** [comment]
> Here is the german localization:
> ...

I think you should create another PR for this.

**Comment by CodeTappert** [comment]
> > Here is the german localization:
> > ...
> 
> I think you should create another PR for this.

Ok. Then please tell me when this pr gets merged so i can open the new one

**Comment by Somod1** [comment]
zh-cn version is complete, wait for this pr to be merged.

**Comment by Somod1** [comment]
zh-cn version is done, wait for this pr to be merged.

**Review by bennybroseph — APPROVED** [review]

**Comment by imfanhua** [comment]
@CodeTappert @Somod1 This PR is merged! You may need to add the missing QUICK CLAW translation entry.

**Comment by imfanhua** [comment]
@CodeTappert @Somod1 This PR has been merged! You may need to add the missing QUICK CLAW translation entry.

**Pull request merged**